source/features/latest-tag-button.tsx

```diff
@@ -18,67 +30,101 @@ const getLatestTag = cache.function(async (): Promise<string | false> => {
-	if (tags.length === 0) {
-		return false;
+	if (repository.refs.nodes.length === 0) {
+		return {
+			latestTag: false,
+			isUpToDate: false
+		};
 	}
 
-	// If all tags are plain versions, parse them
-	if (tags.every(tag => /^[vr]?\d/.test(tag))) {
-		return tags.sort(compareVersions).pop()!;
+	const tags: Tag[] = repository.refs.nodes.map((node: AnyObject) => ({
+		name: node.name,
+		commit: node.tag.commit.oid
+	}));
+
+	// Default to the first tag in the (reverse chronologically-sorted) list
+	let [latestTag] = tags;
+
+	// If all tags are plain versions, sort them as versions
+	if (tags.every(tag => /^[vr]?\d/.test(tag.name))) {
+		latestTag = tags.sort((tag1, tag2) => compareVersions(tag1.name, tag2.name)).pop()!;
 	}
 
-	// Otherwise just use the latest
-	return tags[0];
+	return {
+		latestTag: latestTag.name,
+		isUpToDate: latestTag.commit === repository.defaultBranchRef.target.oid
+	};
 }, {
 	expiration: 1,
 	cacheKey: () => __featureName__ + ':' + getRepoURL()
 });
 
-function getTagLink(latestRelease: string): HTMLAnchorElement {
-	const link = <a className="btn btn-sm btn-outline tooltipped tooltipped-ne ml-2">{tagIcon()}</a> as unknown as HTMLAnchorElement;
+async function init(): Promise<false | void> {
+	const {latestTag, isUpToDate} = await getRepoPublishState();
+	if (!latestTag) {
+		return false;
+	}
 
-	const currentBranch = getCurrentBranch();
+	const breadcrumb = await elementReady('.breadcrumb');
+	if (!breadcrumb) {
+		return;
+	}
 
-	if (currentBranch === latestRelease) {
-		link.classList.add('disabled');
-		link.setAttribute('aria-label', 'You’re on the latest release');
+	const currentBranch = getCurrentBranch();
+	let href: string;
+	if (isRepoRoot()) {
+		href = `/${getRepoURL()}/tree/${latestTag}`;
 	} else {
-		if (isRepoRoot()) {
-			link.href = `/${getRepoURL()}/tree/${latestRelease}`;
-		} else {
-			link.href = replaceBranch(currentBranch, latestRelease);
-		}
-
-		link.setAttribute('aria-label', 'Visit the latest release');
-		link.append(' ', <span className="css-truncate-target">{latestRelease}</span>);
+		href = replaceBranch(currentBranch, latestTag);
 	}
 
-	return link;
-}
+	const link = (
+		<a className="btn btn-sm btn-outline tooltipped tooltipped-ne ml-2" href={href}>
+			{tagIcon()}
+		</a>
+	);
 
-async function init(): Promise<false | void> {
-	const [breadcrumbs, latestTag] = await Promise.all([
-		elementReady('.breadcrumb'),
-		getLatestTag()
-	]);
+	breadcrumb.before(link);
+	if (currentBranch !== latestTag) {
+		link.append(' ', <span className="css-truncate-target">{latestTag}</span>);
+	}
 
-	if (!breadcrumbs || !latestTag) {
-		return false;
+	if (currentBranch === latestTag || isUpToDate) {
+		link.setAttribute('aria-label', 'You’re on the latest release');
+		link.classList.add('disabled');
+		return;
 	}
 
-	breadcrumbs.before(getTagLink(latestTag));
+	const defaultBranch = await getDefaultBranch();
+	if (currentBranch === defaultBranch) {
```

Comment:
In here we could trigger an extra request to find out exactly how many commits HEAD is behind.

```js
const getDifference = cache.function(async (): Promise<string> => {
	const tagPage = await fetchDom(`/${getRepoURL()}/releases/tag/${latestTag}`);
	// If it exists, this text is "4 commits to master since this tag"
	return select('.release-header relative-time + a[href*="/compare/"]')!.textContent.replace(/\D/g, '');
}
```

```js
 	if (currentBranch === defaultBranch) { 
		const aheadBy = await getDifference();
		// set label and <sup>+${aheadBy}</sup>
	}
```

Reply:
That looks (and behaves) quite nicely @fregante - thanks!
It now looks like this:
![xCommitsAhead](https://user-images.githubusercontent.com/6058517/74824099-abc25800-52ff-11ea-8874-55a33f9074e6.png)


Note: I had to change the query to get the tag OID, as tags don't always have explicit commits attached.  
See for example this query:
```
{
  repository(name: "caprine", owner: "sindresorhus") {
    refs(first: 5, refPrefix: "refs/tags/", orderBy: {field: TAG_COMMIT_DATE, direction: DESC}) {
      nodes {
        name
        tag: target {
          oid
          ... on Tag {
            commit: target {
              abbreviatedOid
              oid
              id
            }
          }
        }
      }
    }
  }
}

```
Tag 2.42.1 doesn't have a commit attached to it, so the tag-OID is the same as the commit-oid.
Other tags do have a separate tag-oid and commit-oid.

Not sure what the difference is, why some tags don't have an explicit commit? 
I changed the logic to accept both, depending on whether the commit is available

Reply:
Indeed! Thanks for the change

https://mijingo.com/blog/two-types-of-git-tags